platform/ios/detect.py

```diff
@@ -38,6 +38,9 @@ def get_flags():
     return [
         ("tools", False),
         ("use_volk", False),
+        # Disable by default even if production is set, as it makes linking in Xcode
+        # on exports very slow and that's not what most users expect.
+        ("lto", "none"),
```

Comment:
The idea here is to still all specifying `lto=<thin|full>` if wanted, but make sure that `production=yes` will not enable LTO unless requested explicitly.

Seems to work on Linux if I add the same to `get_flags`, but didn't test this for iOS specifically.